### ISSUE TITLE
context deadline exceeded error when querying data from mertic in new relic

### ISSUE THREAD

**Issue by dkvolam** [newrelic/newrelic-snowflake-integration#33]
HI team,

I have done all the integration set up with snowflake and i was able to query the data in new relic for few days, now am seeing context deadline exceeded error and not able to see the data in new relic.

please help.

**Comment by dkvolam** [comment]
I received the below message from new relic team:
Please someone help on this...

============================
Thanks for reaching out to New Relic! My name is Aaron and I'm on the infrastructure support team, and would be happy to assist you with this request.

Context Deadline Exceeded errors typically indicated either a communication issue between the host and our endpoint, or it can also be related to the amount of data attempting to be ingested and timing out before it has been completely sent / received. You can try adjusting your timeouts in the infrastructure agent configuration, or the interval at which data is being collected.

Are you connecting through a proxy of any kind? You may want to check to ensure there isn't a communication issue / delay between the proxy too.

Since our Snowflake integration is currently a community project, it is [Out of Scope for Technical Support](https://urldefense.com/v3/__https://docs.newrelic.com/docs/licenses/license-information/general-usage-licenses/global-technical-support-offerings/*scope-of-support__;Iw!!Hit2Ag!25q2WFG24GhO8wzlGsQZgeTv5JIGDlzHkOSGMWL-HyGPoUL8VoaZZkepTwHz_U-u8pgE2YayxrbR8pOOO7uIISsWYUAEtgsP1WGhI4s$), and the best option is to [file an issue within the Snowflake repository](https://urldefense.com/v3/__https://github.com/newrelic/newrelic-snowflake-integration*support__;Iw!!Hit2Ag!25q2WFG24GhO8wzlGsQZgeTv5JIGDlzHkOSGMWL-HyGPoUL8VoaZZkepTwHz_U-u8pgE2YayxrbR8pOOO7uIISsWYUAEtgsPLY6H5Iw$) directly and one of our engineers can look into the issue there.

I will mark this ticket as 'Solved' so you do not receive any of the auto-messages from our Support System -- Please feel free to re-open within 30 days if you need any further assistance!

============================================

**Comment by adlangx** [comment]
Having the same issue.

**Comment by jimhoonan** [comment]
I think that means the queries are timing out. Default timeout time is 10 seconds. You can increase the query timout time in the flexConfig yaml like so

`        - name: snowflakeAccountMetering
          # Enables remote monitoring which prevents adding host metadata
          entity: snowflake
          event_type: SnowflakeAccount
          custom_attributes:
            metric_type: snowflake.billing
          commands:
            - run: /app/snowflakeintegration-linux /app/config.yaml /app/queries/account_metering.sql
              timeout: 60000
        - name: snowflakeAutomaticClusteringUsage
          entity: snowflake
          event_type: SnowflakeAccount
          custom_attributes:
            metric_type: snowflake.automatic_clustering
          commands:
            - run: /app/snowflakeintegration-linux /app/config.yaml /app/queries/automatic_clustering_usage.sql
              timeout: 60000`

**Comment by adlangx** [comment]
Yeah, our issue was that the table_storage_usage.sql query was hanging. It was running for 10 minutes when I found it and killed it. We finally just removed it from the queries list and now the integration is working.

**Comment by rashmi050918** [comment]
i am also facing same issue as context deadline exceeded. For testing purpose i have just kept 1 event , snowflake_billing ,also timeout is increased., still no data reported. can you please help

**Comment by suzuki-eiji** [comment]
I have the same issue.

Setting the timeout under commands doesn’t seem to work:
>         - name: snowflakeAutomaticClusteringUsage
>          entity: snowflake
>          event_type: SnowflakeAccount
>          custom_attributes:
>            metric_type: snowflake.automatic_clustering
>          commands:
>            - run: /app/snowflakeintegration-linux /app/config.yaml /app/queries/automatic_clustering_usage.sql
>              timeout: 60000

Has anyone found a solution?
